-1 
Tested at: 733a6c6b465c7205b41473f0f646d163a0c75a6c
 I found a compilation error while trying to compile with clang: 
I used this command: 
 scram b vclean && scram build -k -j 8 USER_CXXFLAGS='-fsyntax-only' COMPILER='llvm compile'


 <pre>          ^
/afs/cern.ch/cms/sw/ReleaseCandidates/vol1/slc6_amd64_gcc491/cms/cmssw-patch/CMSSW_7_5_X_2015-04-07-2300/src/TrackingTools/TrajectoryState/interface/BasicTrajectoryState.h:89:19: note: overridden virtual function is here
  virtual pointer clone() const=0;
                  ^
>> Compiling edm plugin /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_5_X_2015-04-07-2300/src/RecoParticleFlow/PFSimProducer/plugins/ConvBremSeedProducer.cc 
clang: error: unknown argument: '-fipa-pta'
clang: error: unknown argument: '-fipa-pta'
gmake: *** [tmp/slc6_amd64_gcc491/src/RecoParticleFlow/PFSimProducer/plugins/RecoParticleFlowPFTrackingSimPlugins/ConvBremSeedProducer.o] Error 1
>> Compiling edm plugin /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_5_X_2015-04-07-2300/src/RecoParticleFlow/PFSimProducer/plugins/modulesSim.cc 
clang: error: unknown argument: '-fipa-pta'
In file included from /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_5_X_2015-04-07-2300/src/Calibration/IsolatedParticles/src/MatrixECALDetIds.cc:6:
</pre> 
 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-8654/4111/summary.html

--- a/DataFormats/TrackerRecHit2D/interface/SiTrackerGSMatchedRecHit2D.h
+++ b/DataFormats/TrackerRecHit2D/interface/SiTrackerGSMatchedRecHit2D.h
@@ -63,13 +63,14 @@ public:
  ClusterRef const& cluster() const { return cluster_;}
   void setClusterRef(const ClusterRef &ref) { cluster_  = ref; }
  
+  void setEeId(uint32_t eeId){eeId_ = eeId;}
 
   virtual bool sharesInput( const TrackingRecHit* other, SharedInputType what) const;
  
 private:
   int const simhitId_;
   int const simtrackId_;
-  uint32_t const eeId_;
+  uint32_t eeId_;
   ClusterRef cluster_;
   int const pixelMultiplicityAlpha_;
   int const pixelMultiplicityBeta_;

--- a/DataFormats/TrackerRecHit2D/interface/SiTrackerGSRecHit2D.h
+++ b/DataFormats/TrackerRecHit2D/interface/SiTrackerGSRecHit2D.h
@@ -48,6 +48,8 @@ public:
   ClusterRef const& cluster() const { return cluster_;}
   void setClusterRef(const ClusterRef &ref) { cluster_  = ref; }
   
+  void setEeId(uint32_t eeId){eeId_ = eeId;}
+
   virtual bool sharesInput( const TrackingRecHit* other, SharedInputType what) const {return false;}
   
  private:

--- a/SimGeneral/MixingModule/plugins/Adjuster.cc
+++ b/SimGeneral/MixingModule/plugins/Adjuster.cc
@@ -43,5 +43,16 @@ void doTheOffset(int bunchSpace, int bcr, std::vector<PCaloHit>& calohits, unsig
     item.setTime(item.time() + timeOffset);
   }
 }
-}
-}
+
+  void doTheOffset(int bunchSpace, int bcr, TrackingRecHitCollection & trackingrechits, unsigned int evtNr, int vertexOffset) {
+
+    EncodedEventId id(bcr,evtNr);
+    for (auto it = trackingrechits.begin();it!=trackingrechits.end();++it) {
+      SiTrackerGSMatchedRecHit2D * rechit = dynamic_cast<SiTrackerGSMatchedRecHit2D*>(&(*it));
+      if(rechit){
+	rechit->setEeId(id.rawId());
+      }
+    }
+  }
+} // end namespace detail
+} // end namespace edm                   

--- a/SimGeneral/MixingModule/plugins/Adjuster.h
+++ b/SimGeneral/MixingModule/plugins/Adjuster.h
@@ -53,13 +53,13 @@ namespace edm {
     void doTheOffset(int bunchspace, int bcr, std::vector<SimVertex>& product, unsigned int eventNr, int vertexOffset);
     void doTheOffset(int bunchspace, int bcr, std::vector<PCaloHit>& product, unsigned int eventNr, int vertexOffset);
     void doTheOffset(int bunchspace, int bcr, std::vector<PSimHit>& product, unsigned int eventNr, int vertexOffset);
+    void doTheOffset(int bunchspace, int bcr, TrackingRecHitCollection & product, unsigned int eventNr, int vertexOffset);
   }
 
   template<typename T>
   void  Adjuster<T>::doOffset(int bunchspace, int bcr, const EventPrincipal &ep, ModuleCallingContext const* mcc, unsigned int eventNr, int vertexOffset) {
-    std::shared_ptr<Wrapper<std::vector<T> > const> shPtr = getProductByTag<std::vector<T> >(ep, tag_, mcc);
     if (shPtr) {
-      std::vector<T>& product = const_cast<std::vector<T>&>(*shPtr->product());
+      T& product = const_cast<T&>(*shPtr->product());
       detail::doTheOffset(bunchspace, bcr, product, eventNr, vertexOffset);
     }
   }

--- a/SimGeneral/MixingModule/plugins/BuildFile.xml
+++ b/SimGeneral/MixingModule/plugins/BuildFile.xml
@@ -19,6 +19,7 @@
 <use   name="CondFormats/DataRecord"/>
 <use   name="CondFormats/RunInfo"/>
 <use   name="CondCore/DBOutputService"/>
+<use   name="DataFormats/TrackerRecHit2D"/>
 <library   file="*.cc" name="SimGeneralMixingModulePlugins">
   <flags   EDM_PLUGIN="1"/>
 </library>

